tests/integration/dhcp_test.py

```diff
@@ -192,6 +193,126 @@ def test_dhcp_for_bond_with_ip_address_and_slave(dhcp_env,
     assertlib.assert_state(desired_state)
 
 
+def test_ipv4_dhcp_ignore_gateway(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-gateway'] = True
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure dhcp gateway got ignored.
+
+
+def test_ipv4_dhcp_ignore_dns(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-dns'] = True
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support DNS query, need to make
+    # sure dhcp DNS entries got ignored.
+
+
+def test_ipv4_dhcp_ignore_routes(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-routes'] = True
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
```

Comment:
Is there a flag on the `dhclient` command-line that will should this has been added? Or is this a config file generated by NM? Can we check it as a temporary assertion?

BTW: I would prefer not to sign on the TODO with names, as someone else may take it in the future. The one who wrote the TODO is already saved by `git`.

Reply:
The nm(version 1.14.5) I am using does not invoke `dhclient` command or other commands.
And I failed to find a simple to verify this option.

OK. I will remove the TODO signing.

Reply:
This is very strange, on CentOS 7.5 `ps -ef | grep dhclient` shows it is running.
What changed?

Reply:
In manpage of `networkmanager.conf`:

```
dhcp
   This key sets up what DHCP client NetworkManager will use. Allowed values
   are dhclient, dhcpcd, and internal. The dhclient and dhcpcd options require
   the indicated clients to be installed. The internal option uses a built-in
   DHCP client which is not currently as featureful as the external clients.

   If this key is missing, it defaults to internal. It the chosen plugin is not
   available, clients are looked for in this order: dhclient, dhcpcd, internal.
```

I guess since certain version, NetworkManager default internal dhcp client.